### DISCUSSION TITLE
User interaction with layer (e.g. to edit a polygon)

### DISCUSSION THREAD

**Discussion by JonasVautherin** [maplibre/maplibre-native#856] [Q&A]
I see that one can draw lines/polygons on a layer, e.g. like [described here](https://docs.maptiler.com/maplibre-gl-native-android/android-geojson-overlay/). Seems like a lower-level API than an annotation, which is fine :+1:.

I believe I get how I could draw a polyline/polygon given a list of global geo coordinates. But what if I wanted the user to be able to edit the polygon? Say the user can define an area on the map by moving (drag-and-drop) the corners of that polygon on the map.

Would that be possible? I guess it would require a way to transform the local coordinates of the touch (given by a `MotionEvent` maybe?) to a global geo coordinate for the map, right?

**Comment by mr337** [comment]
I don't know if this is still an issue but we ran into the exact same problem, hopping to find a better solution.

Our only solution was to open up a webview and load the JS version of maplibre and follow this https://maplibre.org/maplibre-gl-js-docs/example/mapbox-gl-draw/.

It does work but isn't great. For example if a user doesn't have an existing element they want to modify, such as creating a new element, we would like to use their current GPS location. On Android this isn't too much of a problem, but with iOS it will then re-prompt the user using a rough dialog asking for GPS permission again.

**Comment by JonasVautherin** [comment]
Actually we managed to solve it and released our solution as a small library :blush:: https://github.com/ramani-maps/ramani-maps/

**Reply by mr337** [reply]
Glad to hear, we'll kick the tires on the new library shortly!

**Comment by CodeK1988** [comment]
@JonasVautherin hi mate, i m android dev.When creating a symbol using SymbolManager, I noticed that fast dragging is unresponsive and has a high latency. Is there a solution? What I want is to edit lines by dragging and dropping Markers (symbols); the line dragging function in the map plugin does not meet my needs. thanks

**Reply by JonasVautherin** [reply]
Which map plugin do you mean?

**Reply by CodeK1988** [reply]
https://github.com/maplibre/maplibre-plugins-android @JonasVautherin
